OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -0,0 +1,79 @@
+﻿using OWML.Common;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace OWML.ModHelper.Interaction
+{
+    public class ModInteraction : IModInteraction
+    {
+        private readonly IList<IModBehaviour> _modList;
+        private Dictionary<IModBehaviour, List<IModBehaviour>> _dependantDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+        private Dictionary<IModBehaviour, List<IModBehaviour>> _dependencyDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+
+        public ModInteraction(IList<IModBehaviour> list)
+        {
+            _modList = list;
+
+            RegenDicts();
+        }
+
+        private void RegenDicts()
+        {
+            _dependantDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+            _dependencyDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+            foreach (var mod in _modList)
+            {
+                var dependants = new List<IModBehaviour>();
+                var dependencies = new List<IModBehaviour>();
+                foreach (var dependency in _modList)
+                { 
+                    if (dependency.ModHelper.Manifest.Dependencies.Any() && dependency.ModHelper.Manifest.Dependencies.Contains(mod.ModHelper.Manifest.UniqueName))
+                    {
+                        dependants.Add(dependency);
+                    }
+
+                    if (mod.ModHelper.Manifest.Dependencies.Contains(dependency.ModHelper.Manifest.UniqueName))
+                    {
+                        dependencies.Add(dependency);
+                    }
+                }
+                _dependantDict.Add(mod, dependants);
+                _dependencyDict.Add(mod, dependencies);
+            }
+        }
+
+        public IList<IModBehaviour> GetDependants(string dependencyUniqueName)
+        {
+            RegenDicts();
+            return _dependantDict.Where(x => x.Key.ModHelper.Manifest.UniqueName == dependencyUniqueName).FirstOrDefault().Value;
```

Comment:
Seems like we're using dictionaries to store the dependencies, but then we're not taking advantage of the fact that they are dictionaries. I think we can change the dependency and dependant dictionary definition to `Dictionary<string, List<IModBehaviour>>`, where the string is the uniqueName. So when you need to get a dependency / dependant you do `_dependantDict[dependencyUniqueName]`.